packages/@react-stately/datepicker/src/useTimeFieldState.ts

```diff
@@ -46,8 +46,8 @@ export function useTimeFieldState<T extends TimeValue = TimeValue>(props: TimeFi
   let placeholderDate = useMemo(() => {
     let valueTimeZone = v && 'timeZone' in v ? v.timeZone : undefined;
 
-    return valueTimeZone || defaultValueTimeZone ? toZoned(convertValue(placeholderValue), valueTimeZone || defaultValueTimeZone) : convertValue(placeholderValue);
-  }, [placeholderValue]);
+    return (valueTimeZone || defaultValueTimeZone) && placeholderValue ? toZoned(convertValue(placeholderValue), valueTimeZone || defaultValueTimeZone) : convertValue(placeholderValue);
```

Comment:
I'm surprised that the TS strict null check doesn't complain about convertValue possibly returning null and that being passed into toZoned previously, shrug